### ISSUE TITLE
Syntax highlighting of all types is broken in VSCode

### ISSUE THREAD

**Issue by Peter-Juhasz** [Microsoft/TypeScript#18038]
- VSCode Version: Code 1.15.1 (41abd21afdf7424c89319ee7cb0445cc6f376959, 2017-08-16T18:05:38.342Z)
- OS Version: Windows_NT ia32 10.0.15063
- TypeScript SDK installed: 2.4.1.0

Expected (Visual Studio 2017):

![image](https://user-images.githubusercontent.com/9047283/29706122-f902652e-8980-11e7-8e9a-68b46c0f1cdb.png)

Actual (Visual Studio Code):

![image](https://user-images.githubusercontent.com/9047283/29706682-fcd128d2-8982-11e7-9ee6-7973fe543b2e.png)

Steps to Reproduce:

```ts
let a: string = "";
```

produces this:

![image](https://user-images.githubusercontent.com/9047283/29706173-1963f8a0-8981-11e7-846b-ea0865253022.png)

Reproduces without extensions: Yes

**Comment by ikatyang** [comment]
I think there should be something wrong with your theme or extension, not TypeScript. Can you give a try with `code --disable-extensions` or changing the theme?

(vscode 1.15.1, theme: One Dark Pro)

![image](https://user-images.githubusercontent.com/8341033/29706467-84876b2a-89b4-11e7-9c79-4db1ad156707.png)

```ts
/**
 * Represents a view for sorting, filtering, navigating a data collection.
 */
export class CollectionView<TModel> {
    public constructor(
        public data: TModel[],
        sortExperssion?: keyof TModel
    ) {
        this.refresh();

        if (sortExperssion)
            this.sort(sortExpression);
    }

    public filteredView: TModel[];
    public predicate: ((item: TModel) => boolean) | null = null;
}
```

**Comment by Peter-Juhasz** [comment]
Hey @ikatyang, thanks for your quick response. Yes, I did. This is a screenshot with all of my extensions disabled:

![image](https://user-images.githubusercontent.com/9047283/29707010-09ac7830-8984-11e7-864a-b4a29a80f673.png)

I am using the "Light (Visual Studio)" theme, but I can see the same issues with the dark theme as well.

These are my settings:
```json
{
    "editor.glyphMargin": true,
    "editor.formatOnType": true,
    "editor.roundedSelection": false,
    "php.validate.executablePath": "C:\\Program Files\\PHP\\php.exe",
    "editor.selectionHighlight": false,
	"editor.fontSize": 16,
    "editor.wordBasedSuggestions": false,
    "editor.snippetSuggestions": "inline",
    "editor.cursorBlinking": "smooth",
    "editor.mouseWheelZoom": true,
    "editor.renderLineHighlight": "none",
    "editor.formatOnPaste": true,
    "workbench.colorTheme": "Visual Studio Dark",
    "workbench.iconTheme": "vs-minimal",
    "extensions.autoUpdate": true,
    "php.validate.run": "onType",
    "typescript.referencesCodeLens.enabled": true,
    "typescript.implementationsCodeLens.enabled": true,
    "editor.acceptSuggestionOnCommitCharacter": true,
    "editor.dragAndDrop": true,
    "files.associations": {
        "*.module": "php",
        "*.inc": "php"
    },
    "workbench.sideBar.location": "right",
    "files.autoSave": "afterDelay",
    "files.autoSaveDelay": 500,
    "editor.cursorStyle": "line-thin",
    "window.menuBarVisibility": "toggle",
    "editor.showFoldingControls": "always",
    "editor.renderIndentGuides": false,
    "git.enableSmartCommit": true,
    "git.confirmSync": false,
    "editor.minimap.renderCharacters": false,
    "diffEditor.renderIndicators": false,
    "javascript.referencesCodeLens.enabled": true
}
```

**Comment by Peter-Juhasz** [comment]
Oh wait, with the "Light+" theme some of the issues are resolved.
 - the lambda arrow is still highlighted as a keyword
 - while some keywords like (boolean, string) are highlighted as user-defined types instead keywords

![image](https://user-images.githubusercontent.com/9047283/29707525-d2a981aa-8985-11e7-87cf-3ace5272cc0c.png)

**Comment by kitsonk** [comment]
You might want to make sure what actual tokens are being recognised.

In vscode it is accessed via the command _Developer: Inspect TM Scopes_ which will then provide you tokenisation information for everything and rule out an issue with the theme as well as provide what the TypeScript services actually thinks something is.

![inspect](https://user-images.githubusercontent.com/1282577/29708442-8a07986a-8980-11e7-81e8-7fcc4f7a13b2.gif)

**Comment by ikatyang** [comment]
See https://github.com/Microsoft/vscode/blob/master/extensions/theme-defaults/themes/light_vs.json#L196

```json
{
    "scope": "storage.type",
    "settings": {
        "foreground": "#0000ff"
    }
}
```

![image](https://user-images.githubusercontent.com/8341033/29708557-93ffe940-89bb-11e7-8074-de607c492bed.png)

**Comment by Peter-Juhasz** [comment]
I tried to uninstall/reinstall but didn't help. I checked the theme files manually and I think some rules may be missing from the "Light (Visual Studio)" theme.

Thank you very much for showing me this developer tool, I can find all the classifications and customize them in the user settings file, so I can make the look the same as VS as I like. It's awesome, thank you.

![image](https://user-images.githubusercontent.com/9047283/29709045-2cb8ff7c-898b-11e7-9619-4c089f772dab.png)

By the way, what is your opinion about making VS and VS Code light theme look the same? I would be happy to create a pull request to fine tune some of these rules to match VS.
